### PR TITLE
Link media to multiple structural elements

### PR DESCRIPTION
**Pull request by oliver-stoehr**
MediaUnits can be linked to multiple IncludedStructuralElements to represent multiple logical structures on the same media element (e.g. page).

### CODE REVIEW

File: Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java [hunk 1]
```diff
@@ -1114,4 +1122,114 @@ private IncludedStructuralElement getTreeNodeStructuralElement(TreeNode treeNode
...
+
+    /**
+     * Get the index of this StructureTreeNode's MediaUnit out of all MediaUnits
+     * which are assigned to more than one IncludedStructuralElement.
+     *
+     * @param treeNode object to find the index for
+     * @return index of the StructureTreeNode's MediaUnit if present in the List of several assignments, or -1 if not present in the list.
+     */
+    public int getMultipleAssignmentsIndex(StructureTreeNode treeNode) {
+        if (Objects.nonNull(treeNode.getDataObject()) && treeNode.getDataObject() instanceof View
```
Comment (Beacze) [inline Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java]:
```suggestion
        if (treeNode.getDataObject() instanceof View
```

instanceof already checks against null

File: Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java [hunk 2]
```diff
@@ -1114,4 +1122,114 @@ private IncludedStructuralElement getTreeNodeStructuralElement(TreeNode treeNode
...
+
+    /**
+     * Check if the selected Node's MediaUnit is assigned to several IncludedStructuralElements.
+     *
+     * @return {@code true} when the MediaUnit is assigned to more than one logical element
+     */
+    public boolean isAssignedSeveralTimes() {
+        if (Objects.nonNull(selectedLogicalNode) && selectedLogicalNode.getData() instanceof  StructureTreeNode) {
+            StructureTreeNode structureTreeNode = (StructureTreeNode) selectedLogicalNode.getData();
+            if (Objects.nonNull(structureTreeNode.getDataObject()) && structureTreeNode.getDataObject() instanceof View) {
```
Comment (Beacze) [inline Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java]:
```suggestion
            if (structureTreeNode.getDataObject() instanceof View) {
```
instanceof already checks against null

File: Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java [hunk 3]
```diff
@@ -1114,4 +1122,114 @@ private IncludedStructuralElement getTreeNodeStructuralElement(TreeNode treeNode
...
+    }
+
+    /**
+     * Check if the selected Node's MediaUnit can be assigned to the next logical element in addition to the current assignment.
+     * @return {@code true} if the MediaUnit can be assigned to the next IncludedStructuralElement
+     */
+    public boolean isAssignableSeveralTimes() {
+        if (Objects.nonNull(selectedLogicalNode) && selectedLogicalNode.getData() instanceof  StructureTreeNode) {
+            StructureTreeNode structureTreeNode = (StructureTreeNode) selectedLogicalNode.getData();
+            if (Objects.nonNull(structureTreeNode.getDataObject())
```
Comment (Beacze) [inline Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java]:
```suggestion
            if (structureTreeNode.getDataObject() instanceof View) {
```

File: Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java [hunk 4]
```diff
@@ -1114,4 +1122,114 @@ private IncludedStructuralElement getTreeNodeStructuralElement(TreeNode treeNode
...
+                    && structureTreeNode.getDataObject() instanceof View) {
```
Comment (Beacze) [inline Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java]:
```suggestion

```

**Comment by Erikmitk** [comment]
Please resolve conflicts.

**Pull request merged**